source/isaaclab/isaaclab/controllers/null_space_posture_task.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import numpy as np
+
+import pinocchio as pin
+from pink.configuration import Configuration
+from pink.tasks import Task
+from pink.utils import get_root_joint_dim
+
+
+class NullSpacePostureTask(Task):
```

Comment:
Do you plan to add more of these tasks in the future? If so, I suggest making a separate folder called "pink_tasks" and have it in there. Then later on, other tasks can be added easily.

Reply:
Also please update the API docs to include these in the doc generation: https://github.com/michaellin6/IsaacLab/blob/1bd63bffc5f54bfadb1c927f25bb4eaa760d71bf/docs/source/api/lab/isaaclab.controllers.rst